lxd/api_internal.go

```diff
@@ -393,6 +400,56 @@ func internalContainerOnStop(d *Daemon, r *http.Request) response.Response {
 	return response.EmptySyncResponse
 }
 
+func internalVirtualMachineOnResize(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	// Get the instance ID.
+	instanceID, err := strconv.Atoi(mux.Vars(r)["instanceRef"])
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	// Get the devices list.
+	devices := request.QueryParam(r, "devices")
+	if devices == "" {
+		return response.BadRequest(fmt.Errorf("Resize hook requires a list of devices"))
+	}
+
+	// Load by ID.
+	inst, err := instance.LoadByID(s, instanceID)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Update the local instance.
+	for _, dev := range strings.Split(devices, ",") {
+		fields := strings.SplitN(dev, ":", 2)
+		if len(fields) != 2 {
+			return response.BadRequest(fmt.Errorf("Invalid device/size tuple: %s", dev))
+		}
+
+		size, err := strconv.ParseInt(fields[1], 16, 64)
```

Comment:
Is the `size` really expressed in base 16?

Reply:
Nice catch!